sycl/include/CL/sycl/event.hpp

```diff
@@ -132,17 +132,34 @@ class __SYCL_EXPORT event {
   /// \return a native handle, the type of which defined by the backend.
   template <backend BackendName>
   __SYCL_DEPRECATED("Use SYCL 2020 sycl::get_native free function")
-  auto get_native() const ->
-      typename detail::interop<BackendName, event>::type {
+  detail::enable_if_t<
+      (BackendName != backend::opencl),
+      typename detail::interop<BackendName, event>::type> get_native() const {
     return reinterpret_cast<typename detail::interop<BackendName, event>::type>(
         getNative());
   }
+  /// \return a native handle vector for opencl according to specification.
+  template <backend BackendName>
+  __SYCL_DEPRECATED("Use SYCL 2020 sycl::get_native free function")
+  detail::enable_if_t<(BackendName == backend::opencl),
+                      std::vector<cl_event>> get_native() const {
```

Comment:
shouldn't the type returned by enable_if_t  use `backend_return_t<>` rather than coupling tightly by specifying `std::vector<cl_event>` directly?

Reply:
yes, backend_return_t<> is preferred

Reply:
Fixed